Entities/Structures/Common/AddTilesBySector.as

```diff
@@ -34,7 +11,9 @@ void AddTilesBySectorWithWindow(Vec2f ul, Vec2f lr, string sectorName, TileType
 		{
 			while (tpos.y < lr.y)
 			{
-				if (map.getSectorAtPosition(tpos, sectorName) !is null && map.getTile(tpos).type != omitTile && tpos != center)
+				if (map.getSectorAtPosition(tpos, sectorName) !is null 
+					&& map.getTile(tpos).type != omitTile
+					&& !(tpos == center && hasWindow))
```

Comment:
Rearranging this set of conditions like so makes it a bit more readable in my opinion. Feel free to change this if you agree.
```suggestion
					&& (!hasWindow || tpos != center))
```